### PR TITLE
KMM-9763 :: Feature :: Update dependencies

### PR DESCRIPTION
**Pull request by franmontiel**
This PR updates dependencies, the most important updates are:
- Kotlin to 1.7.10
- Ktor to 2.0.3 

**IMPORTANT**
- After this PR is merged the new native memory management model must be enabled in the projects: https://github.com/JetBrains/kotlin/blob/master/kotlin-native/NEW_MM.md#switch-to-the-new-mm

Note that this PR goes to **feature/mvi** instead of master

**Merge / Pull request information:**

* Asana task link: https://app.asana.com/0/1109863238977521/1202789494669763

### CODE REVIEW

File: harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/memory/InMemoryDataSource.kt [hunk 1]
```diff
@@ -13,16 +12,14 @@ import com.harmony.kotlin.error.notSupportedQuery
 
 class InMemoryDataSource<V> : GetDataSource<V>, PutDataSource<V>, DeleteDataSource {
```
Comment (franmontiel) [inline harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/memory/InMemoryDataSource.kt:13]: With the new memory model IsolateState shouldn't be necessary. 

Removed the fix and still passing the test

File: buildSrc/src/main/java/Dependencies.kt [hunk 2]
```diff
@@ -3,16 +3,16 @@ const val android_min_sdk_version = 21
...
-const val ktor_version = "1.6.8"
-const val serialization_version = "1.3.2"
+const val ktor_version = "2.0.3"
+const val serialization_version = "1.3.3"
 const val coroutines_version = "1.6.3-native-mt"
 const val sql_delight_version = "1.5.3"
-const val kotlinx_datetime_version = "0.3.2"
+const val kotlinx_datetime_version = "0.4.0"
 const val uuid_lib_version = "0.3.1"
 const val isolate_version = "1.2.0"
```
Comment (joselufo) [inline buildSrc/src/main/java/Dependencies.kt:17]: Remove it as you already remove the dependency in the code.

File: harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/memory/InMemoryDataSource.kt (hunk 1 above)
Comment (joselufo) [reply harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/memory/InMemoryDataSource.kt:13]: If it passes the tests, all ok. Another dependency was removed.

**Review by joselufo — CHANGES_REQUESTED** [review]

**Review by jfilowk — APPROVED** [review]

File: buildSrc/src/main/java/Dependencies.kt (hunk 2 above)
Comment (jfilowk) [reply buildSrc/src/main/java/Dependencies.kt]: Removed

**Comment by jfilowk** [comment]
@joselufo merged after fix your comments.

**Pull request merged**